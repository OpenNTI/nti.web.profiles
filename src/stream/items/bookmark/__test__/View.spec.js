import React from 'react';
import renderer from 'react-test-renderer';

import * as TestUtils from '@nti/web-client/test-utils';

import Bookmark from '../View';

const { tearDownTestClient, setupTestClient } = TestUtils;

const mockService = {
	getObject: () => Promise.resolve({ Title: 'Test' }),
	resolveEntity: () => Promise.resolve({ displayName: 'Tim Jones' }),
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(mockService);
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Bookmark', () => {
	beforeEach(onBefore);
	afterEach(onAfter);

	test('renders correctly', async () => {
		const item = {
			creator: 'user4',
			getCreatedTime: () => new Date('2018-08-31'),
			getContextPath: () =>
				Promise.resolve([
					[
						{
							NTIID:
								'tag:nextthought.com,2011-10:IFSTA-Bundle-IFSTA_Book_Aircraft_Rescue_and_Fire_Fighting_Sixth_Edition',
							getPresentationProperties: () => ({
								title:
									'Aircraft Rescue and Fire Fighting Sixth Edition',
							}),
						},
						{
							NTIID:
								'tag:nextthought.com,2011-10:IFSTA-HTML-IFSTA_Book_Aircraft_Rescue_and_Fire_Fighting_Sixth_Edition.section:Engine_Types_and_Applications3',
							Title: 'Engine Types adn Applications',
						},
					],
				]),
			NTIID:
				'tag:nextthought.com,2011-10:user4-OID-0x04e6fa:5573657273:nywS9Pjp8FE',
		};
		const context = {
			getDefaultAssetRoot: () => '',
			NTIID:
				'tag:nextthought.com,2011-10:IFSTA-Bundle-IFSTA_Book_Aircraft_Rescue_and_Fire_Fighting_Sixth_Edition',
			PlatformPresentationResources: [
				{
					Class: 'DisplayablePlatformPresentationResources',
					CreatedTime: 1532453102,
					InheritPlatformName: null,
					'Last Modified': 1527882586,
					PlatformName: 'webapp',
					Version: 1,
					href: '',
				},
			],
		};
		const bookmarkCmp = renderer.create(
			<Bookmark item={item} context={context} />
		);

		jest.runAllTimers();
		await flushPromises();

		const tree = bookmarkCmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
