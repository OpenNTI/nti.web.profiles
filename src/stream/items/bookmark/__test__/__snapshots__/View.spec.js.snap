// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark renders correctly 1`] = `
<div
  className="stream-bookmark"
>
  <div
    className="heading"
  >
    <div
      className="nti-link-to-path"
      to="#"
    >
      <a
        className="username"
        rel="author"
      >
        Tim Jones
      </a>
    </div>
     created a bookmark on 
    <time
      className="nti-text"
      dateTime="2018-08-31T00:00:00+00:00"
    >
      August 31, 2018
    </time>
  </div>
  <div
    className="nti-link-to-path"
    to="/object/user4-OID-0x04e6fa%3A5573657273%3AnywS9Pjp8FE"
  >
    <div
      className="bookmark-content"
    >
      <div
        className="bookmark-charms"
      >
        <div
          className="bookmark-favorite"
        />
      </div>
      <img
        className="bookmark-icon"
        src=""
      />
      <div
        className="bookmark-context"
      >
        <div
          className="nti-content-breadcrumb"
        >
          <div
            className="nti-link-to-path"
            to="/object/user4-OID-0x04e6fa%3A5573657273%3AnywS9Pjp8FE"
          >
            <ul
              className="breadcrumb-list"
            >
              <li
                className="crumb last"
              >
                <span>
                  Engine Types adn Applications
                </span>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="page-title"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;
